File: Insertion Sort.cpp
// Insertion sort in C++

#include <iostream>
using namespace std;

// Function to print an array
void printArray(int array[], int size) {
  for (int i = 0; i < size; i++) {
    cout << array[i] << " ";
  }
  cout << endl;
}

void insertionSort(int array[], int size) {
  for (int i = 1; i < size; i++) {
    int key = array[i];
    int j = i - 1;

    // Compare key with each element on the left of it until an element smaller than it is found.
    while (key < array[j] && j >= 0) {
      array[j + 1] = array[j];
      --j;
    }
    //  key>=array[j]
    array[j + 1] = key;
  }
}

// Driver code
int main() {
  int data[] = {6, 4, 5, 7, 1, 3, 2};
  int size = sizeof(data) / sizeof(data[0]);
  insertionSort(data, size);
  cout << "Sorted array in ascending order:\n";
  printArray(data, size);
}

File: Insertion Sort.py
# -*- coding: utf-8 -*-
"""
Created on Sun Dec 24 12:33:07 2023

@author: User
"""

#Insertion Sort
data = [29, 34, 23, 78, 67, 100, 66, 29, 79, 55, 78, 213, 92, 96, 96, 23]

def InsertionSort(data):
    n = len(data)
    for i in range(1, n):
        key = data[i]
        j = i-1 
        # key < data[j]
        while j >=0 and key < data[j] :
            data[j+1] = data[j]
            j -= 1
            
        # key >= data[j]
        data[j+1] = key
    return data
        
print(InsertionSort(data))